include/dr/details/algorithms.hpp

```diff
@@ -2,17 +2,14 @@
 //
 // SPDX-License-Identifier: BSD-3-Clause
 
-#pragma once
-
-#include <dr/concepts/concepts.hpp>
-#include <dr/shp/algorithms/for_each.hpp>
-
-namespace shp {
+// ALGORITHMS CODE SHARED BETWEEN shp AND mhp
+// DO NOT INCLUDE THIS FILE DIRECTLY
+// INCLUDED BY algorithms.hpp FILES IN mhp AND shp
 
 template <lib::distributed_range R, std::integral T> void iota(R &&r, T value) {
   auto iota_view = rng::views::iota(value, T(value + rng::distance(r)));
 
-  shp::for_each(shp::par_unseq, shp::views::zip(iota_view, r), [](auto &&elem) {
+  for_each(par_unseq, views::zip(iota_view, r), [](auto &&elem) {
```

Comment:
I think I will be changing mhp execution policies, and changing to shp way of just having par_unseq.
I am still fixing some bugs in mhp::zip. Let me know if you see problems